From 4ac134c66591a55bb0cad71d173ac0d71e970658 Mon Sep 17 00:00:00 2001
From: black-desk <me@black-desk.cn>
Date: Fri, 6 Sep 2024 11:51:27 +0800
Subject: [PATCH] include-files: add `pandoc_markdown` support

GFM doesn't support some markdown language extensions that pandoc
enabled by default like grid table.

If someone use pandoc's extended markdown syntax to write markdown
source and publish it on github, it will be a mess.

So we can use `pandoc_markdown` code block to wrap these blocks with
extended syntax, to let github render them like code block, which make
them a lot more readable.

Signed-off-by: black-desk <me@black-desk.cn>
---
 include-files/include-files.lua | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include-files/include-files.lua b/include-files/include-files.lua
index 2499968..3f3c6ba 100644
--- a/include-files/include-files.lua
+++ b/include-files/include-files.lua
@@ -62,6 +62,10 @@ end
 --- Filter function for code blocks
 local transclude
 function transclude (cb)
+  if cb.classes:includes 'pandoc_markdown' then
+    return pandoc.read(cb.text, 'markdown', PANDOC_READER_OPTIONS).blocks
+  end
+
   -- ignore code blocks which are not of class "include".
   if not cb.classes:includes 'include' then
     return
-- 
2.43.0

